packages/block-library/src/buttons/deprecated.js

```diff
@@ -57,7 +59,8 @@ const deprecated = [
 				},
 			},
 		},
-		isEligible: ( { layout } ) => ! layout,
+		isEligible: ( { contentJustification, orientation } ) =>
+			!! contentJustification || !! orientation,
```

Comment:
I still think we shouldn't have `isEligible` here (also [commented about](https://github.com/WordPress/gutenberg/pull/35819#discussion_r742810778) the difference with `social-links`).

This check is handled inside `migrateWithLayout`, no? Have you run into some problems without it?

Reply:
I did try removing it, but `migrateWithLayout` doesn't get called unless we have `isEligible` here. And then trying to edit the layout for existing blocks has unpredictable results.

Reply:
This needs some investigation, but it seems this might be related to the [other deprecation](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/buttons/deprecated.js#L89) that uses `isEligible`. For those we would still need to add the `layout` prop if not exists.

Reply:
Could you tell me more about why we shouldn't have `isEligible` here? As far as I can tell, if we don't use it, the migration won't run. 

The [block handbook](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-deprecation/) says "A deprecation will be tried if the current state of a parsed block is invalid, or if the deprecation defines an isEligible function that returns true". 

In this case, a block with `contentJustification` or `orientation` and a bunch of extra classnames isn't invalid; what happens when it runs through the current save function is the attributes get stripped and the classnames get added as attributes. So, if we want the migration to work (which we do, otherwise it'll be impossible to change the layout of an existing block), we have to use `isEligible`. Or am I missing something here?

> this might be related to the other deprecation that uses isEligible. For those we would still need to add the layout prop if not exists.

I did make sure to add `migrateWithLayout` to the older deprecation too. Is that what you meant?